wasm-thumbnail/src/lib.rs

```diff
@@ -32,11 +32,21 @@ pub extern "C" fn resize_and_pad(
     let slice: &[u8] = unsafe { std::slice::from_raw_parts(pointer, length) };
 
     let mut out: Vec<u8> = Vec::with_capacity(nsize);
-    // Reserve space at the start for length header
-    out.extend_from_slice(&[0, 0, 0, 0]);
 
-    if let Ok(thumbnail_len) = _resize_and_pad(slice, &mut out, nwidth, nheight, nsize, nquality) {
-        out.splice(..4, thumbnail_len.to_be_bytes().iter().cloned());
+
+
+    let mut mquality = nquality;
+
+    while mquality >= 15 {
```

Comment:
So this is definitely an improvement, but I think we could theoretically still not manage to create a small enough image for the size. (maybe for a massive resolution noisy image, with a small `nsize`).

Reply:
Yes, but if we've made ~17 attempts to resize the image and failed, I think it's fair to fall back to previous behavior and have the external program notify the user that a different set of parameters is required, (smaller rez, different image, etc)